plugin/pkg/scheduler/factory/factory.go

```diff
@@ -1137,3 +1150,24 @@ func (p *podPreemptor) UpdatePodAnnotations(pod *v1.Pod, annotations map[string]
 	_, error := p.Client.CoreV1().Pods(podCopy.Namespace).Patch(podCopy.Name, types.MergePatchType, patchData, "status")
 	return error
 }
+
+func (p *podPreemptor) RemoveNominatedNodeAnnotation(pod *v1.Pod) error {
+	podCopy := pod.DeepCopy()
+	if podCopy.Annotations == nil {
+		return fmt.Errorf("pod %v/%v has no annotation", pod.Namespace, pod.Name)
+	}
+	if _, exists := podCopy.Annotations[core.NominatedNodeAnnotationKey]; !exists {
+		return nil
+	}
+	// Note: Deleting the entry from the annotations and passing it Patch() will
```

Comment:
This seems weird. Can you ask the API machinery folks about this?

Reply:
It looks like Path() only make those changes present in the patchData. When there is only one annotation (nominated node name) and we delete it, the annotations become empty and patchData becomes empty. So, patch doesn't make any changes. I will check with the API folks tomorrow.

Reply:
I checked with the API folks and my guess was right. When data is not present in patchData, Patch() interprets it as "no change" to the fields.